tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -0,0 +1,1089 @@
+	wrap_lc_init();
+}
+
+void test_lte_lc_deinit_twice(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	ret = lte_lc_deinit();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	/* this is just to make the tearDown work again */
+	wrap_lc_init();
+}
+
+void test_lte_lc_connect_home_already_registered(void)
+{
+	int ret;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT+CEREG?", "+CEREG: %*u,%hu,%*[^,],\"%x\",", 2);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_NW_REG_REGISTERED_HOME);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint32(0);
+	__wrap_custom_nrf_modem_at_scanf_Ready();
+
+	ret = lte_lc_connect();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_init_and_connect_home_already_registered(void)
+{
+	int ret;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT+CEREG?", "+CEREG: %*u,%hu,%*[^,],\"%x\",", 2);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_NW_REG_REGISTERED_HOME);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint32(0);
+	__wrap_custom_nrf_modem_at_scanf_Ready();
+
+	ret = lte_lc_init_and_connect();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_init_and_connect_async_null(void)
+{
+	int ret;
+
+	ret = lte_lc_init_and_connect_async(NULL);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_lte_lc_normal_success(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", EXIT_SUCCESS);
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", EXIT_SUCCESS);
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+	ret = lte_lc_normal();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_normal_fail1(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", -NRF_ENOMEM);
+	ret = lte_lc_normal();
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_normal_fail2(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", EXIT_SUCCESS);
+	/* enable_notifications */
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", -NRF_ENOMEM);
+	__wrap_nrf_modem_at_cmd_ExpectAnyArgsAndReturn(-NRF_ENOMEM);
+	/* error is ignored */
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CFUN=%d", EXIT_SUCCESS);
+
+
+	ret = lte_lc_normal();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_normal_fail3(void)
+{
+	int ret;
+	char modem_ver[] = "MODEM_FW_VER_1.0\r\nOK";
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CEREG=5", EXIT_SUCCESS);
+	/* enable_notifications */
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT+CSCON=1", -NRF_ENOMEM);
+
+	/* error is ignored but modem FW version is checked */
+	__wrap_nrf_modem_at_cmd_ExpectAndReturn(NULL, 0, "AT+CGMR", EXIT_SUCCESS);
+	__wrap_nrf_modem_at_cmd_IgnoreArg_buf();
+	__wrap_nrf_modem_at_cmd_IgnoreArg_len();
+	__wrap_nrf_modem_at_cmd_ReturnArrayThruPtr_buf(modem_ver, sizeof(modem_ver));
```

Comment:
FYI:
Expected output value for modem fw version is needed because random output crashed in the twister tests in CI.